### PR TITLE
feat: Don't send outbound messages that exceed the max size

### PR DESCRIPTION
**Pull request by mat-if**
## Summary

This adds a check to all outbound messages to ensure they fit within the MAX_MESSAGE_SIZE. This ensures a node is not sending invalid messages, intentionally or not.

## Testing Plan

Unit test

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish/src/network/peers/connections/connection.ts [hunk 1]
```diff
@@ -114,6 +115,47 @@ export abstract class Connection {
...
+          object.type,
+        )} message in state ${this.state.type} ${ErrorUtils.renderError(e)}`,
+      )
+      this.close(e)
+      return false
+    }
+
+    if (sendResult) {
+      this.metrics?.p2p_OutboundTraffic.add(byteCount)
+      this.metrics?.p2p_OutboundTraffic_WebRTC.add(byteCount)
```
Comment (mat-if) [inline ironfish/src/network/peers/connections/connection.ts:152]: 🤦

File: ironfish/src/network/peers/connections/webRtcConnection.ts [hunk 2]
```diff
@@ -197,42 +197,18 @@ export class WebRtcConnection extends Connection {
...
-    }
-
     if (!this.datachannel.isOpen()) {
       this.logger.debug('Datachannel no longer open, closing connection')
       this.close()
       return false
     }
 
-    const data = message.serializeWithMetadata()
-    try {
```
Comment (dguenther) [inline ironfish/src/network/peers/connections/webRtcConnection.ts:220]: I can't say for sure that this try/catch is no longer needed -- might be worth re-adding it unless you checked it out

Comment (mat-if) [reply ironfish/src/network/peers/connections/webRtcConnection.ts:220]: I hoisted it up into the send function: https://github.com/iron-fish/ironfish/pull/3590/files#diff-ed1400fde596eef381f9b8abb8faaa1965e28b481b5f87c829c95c16df06f918R139

Happy to move it back if we prefer - I don't feel strongly one way or the other, I think I moved it just to minimize the amount of differences between the `_send` functions

Comment (dguenther) [reply ironfish/src/network/peers/connections/webRtcConnection.ts:220]: Resolving this, I noticed you boosted it up to the connection level

**Review by dguenther — APPROVED** [review]

**Pull request merged**